CommunityFridgeMapApi/dependencies/python/s3_service.py

```diff
@@ -0,0 +1,106 @@
+import os
+import uuid
+from urllib.parse import urlparse, urlunparse
+import logging
+import boto3
+import botocore
+from botocore.exceptions import ClientError
+from botocore.config import Config
+
+def get_s3_client(env=os.getenv("Environment")):
+    endpoint_url="http://localstack:4566/" if env == "local" else None
+    config = Config(
+        signature_version=botocore.UNSIGNED, # Do not include signatures in s3 presigned-urls.
+    )
+
+    return boto3.client(
+        "s3",
+        config=config,
+        endpoint_url=endpoint_url,
+    )
+
+def translate_s3_url_for_client(url: str, env=os.getenv("Environment")) -> str:
+    """
+    This function translates a S3 url to a client-accessible urls.
+
+    From a lambda function's perspective, local AWS gateway is located at "localstack:4566" (within cfm-network).
+    However, this hostname is not available for the host machine.
+    In order to fetch S3 files from a local browser, we need to use "localhost:4566".
+    """
+    if env == "local":
+        parsed_url = urlparse(url)
+        return urlunparse(parsed_url._replace(netloc="localhost:4566"))
+    return url
+
+
+class S3ServiceException(Exception):
+    pass
+
+
+class S3Service:
+    """
+    Adapter class for persisting binary files in S3 buckets.
+    """
+    def __init__(self):
+        self._client = get_s3_client()
+
+    def idempotent_create_bucket(self, bucket: str):
+        """
+        creates a bucket if there is no existing bucket with the same name.
+            Parameters:
+                bucket: name of the bucket
+        """
+        try:
+            self._client.create_bucket(Bucket=bucket)
```

Comment:
Was getting this error in the deployed lambda. Lets create the S3 bucket from the template.yml file and only call this function if `os.getenv("Environment")` is `local`
```
An error occurred (AccessDenied) when calling the CreateBucket operation: Anonymous access is forbidden for this operation
```

Reply:
That is a good idea. I wish we could just do `samlocal deploy`, but it doesn't work due to the aforementioned lambda layer paywall. Fixed :+1: